daemon/api_users.go

```diff
@@ -241,7 +244,27 @@ func postUsers(c *Command, r *http.Request, user *auth.UserState) Response {
 }
 
 func removeUser(c *Command, username string, opts postUserDeleteData) Response {
-	return NotImplemented("not implemented")
```

Comment:
Should this whole thing be a change with tasks? What if we refresh snapd in the background and decide to reboot after nuking user from the system but keeping the user in state?

Reply:
that's a bigger change TBH and would apply to create and login and everything in this file as well as remove, except for the GET to /v2/users.
Remember snapd does (or tries to do) a clean shutdown, waiting for clients, so in general it'll wait for sync requests to finish before going away.